.dockerignore

```diff
@@ -0,0 +1,16 @@
+/assets/.cache
+/data/favicons/*.png
+/data/favicons/*.jpg
+/data/thumbnails/*.png
+/data/thumbnails/*.jpg
+/data/cache/*.spc
+/data/logs/*.log
+/data/sqlite/*.db
+/public
+user.css
+user.js
+config.ini
+node_modules
+.env
+vendor/
+.php_cs.cache
```

Comment:
According to https://codefresh.io/docker-tutorial/not-ignore-dockerignore-2/, we should also add `.git`.

Reply:
Also probably `/utils`.

Reply:
> According to https://codefresh.io/docker-tutorial/not-ignore-dockerignore-2/, we should also add `.git`.

Updated.

Reply:
> Also probably `/utils`.

You mean we should exclude /utils too from .dockerignore?

Reply:
Yeah, I actually have ton of uncommitted stuff in `/utils`. Though we would also need `!/utils/docker` rule.

Reply:
We need to make a clear distinction about the purpose of this docker image. Two main types are:
- production
- development

Where the production can further be devided into:
- sources come from git clone
- sources come from the current directory (`ADD . /src` in Dockerfile)

My Dockerfile was intended as the production image with sources from git clone. Here is why: it can then be in a separate git repo (such as github.com/SSilence/selfoss-docker) and can be built completely separately yet always resulting in the same source files. It can also be built for some older tag, which is good either for comparison or when master is broken. Using production image with `ADD . /src` is problematic because then even your uncommitted files will be included in the image, which is a problem because you may accidentally push into docker hub something that doesn't exist anywhere in git - this is also how viruses get into docker images. In practice this approach is only used when the build is automated and done in a pipeline on the build server, where we can make sure it checks out a clean workspace from git before the docker build.

So if we keep this image as the production image with sources from git, we don't even need any .dockerignore file at all, because the build will run in a different directory - different git repo.

Reply:
> Hmm, right, but I did it this way because many of the times we want to apply our local repo changes in the docker image itself. And by looking at this use case, it seemed the best choice to me. :-)
But you are absolutely right and then I think that targeting multiple use-cases will be a lot complex. :(
> We need to make a clear distinction about the purpose of this docker image. Two main types are:
> 
> * production
> * development
> 
> Where the production can further be devided into:
> 
> * sources come from git clone
> * sources come from the current directory (`ADD . /src` in Dockerfile)
> 
> My Dockerfile was intended as the production image with sources from git clone. Here is why: it can then be in a separate git repo (such as github.com/SSilence/selfoss-docker) and can be built completely separately yet always resulting in the same source files. It can also be built for some older tag, which is good either for comparison or when master is broken. Using production image with `ADD . /src` is problematic because then even your uncommitted files will be included in the image, which is a problem because you may accidentally push into docker hub something that doesn't exist anywhere in git - this is also how viruses get into docker images. In practice this approach is only used when the build is automated and done in a pipeline on the build server, where we can make sure it checks out a clean workspace from git before the docker build.
> 
> So if we keep this image as the production image with sources from git, we don't even need any .dockerignore file at all, because the build will run in a different directory - different git repo.